### ISSUE TITLE
Integrating google-api-client-android library in Android project causes lint errors

### ISSUE THREAD

**Issue by alipov** [googleapis/google-api-java-client#1539]
#### Environment details

1. General
2. OS type and version: any 
3. Java version: any
4. google-api-client version(s): latest (1.30.9)

#### Steps to reproduce

  1. Add `com.google.api-client:google-api-client-android:1.30.9` as dependency to an Android project.
  2. Try to build a release version.

#### Code example

Official [googlesamples repository](https://github.com/googlesamples/google-services) can be used as an example.  
Clone the repo and try to execute the following task: `./gradlew :signin:app:assembleRelease`
The build fails with lint errors:  
```
> Task :signin:app:lintVitalRelease FAILED
/Users/alex/src/google-services-master/android/signin/app/build.gradle: Error: commons-logging defines classes that conflict with classes now provided by Android. Solutions include finding newer versions or alternative libraries that don't have the same problem (for example, for httpclient use HttpUrlConnection or okhttp instead), or repackaging the library using something like jarjar. [DuplicatePlatformClasses]  
/Users/alex/src/google-services-master/android/signin/app/build.gradle: Error: httpclient defines classes that conflict with classes now provided by Android. Solutions include finding newer versions or alternative libraries that don't have the same problem (for example, for httpclient use HttpUrlConnection or okhttp instead), or repackaging the library using something like jarjar. [DuplicatePlatformClasses]  

   Explanation for issues of type "DuplicatePlatformClasses":
   There are a number of libraries that duplicate not just functionality of
   the Android platform but using the exact same class names as the ones
   provided in Android -- for example the apache http classes. This can lead
   to unexpected crashes.

   To solve this, you need to either find a newer version of the library which
   no longer has this problem, or to repackage the library (and all of its
   dependencies) using something like the jarjar tool, or finally, rewriting
   the code to use different APIs (for example, for http code, consider using
   HttpUrlConnection or a library like okhttp.)
```
If I'll try to follow the suggestions: 

- latest version of a library still has the issue. 
- repackaging the library is probably possible, but is it supported/recommended?  
- rewriting the code to use different APIs - I guess this is a suggestion to library owners, not consumers. 

#### Any additional information below
There's a question regarding [similar issue](https://stackoverflow.com/questions/46989310) in SO, however the answers there are pretty destructive - the main suggestion is to exclude library's compile dependencies. 

What is the recommended/supported way to workaround this issue?

**Comment by alipov** [comment]
I've browsed the sources of this repository, and found that the only place where `org.apache.*` classes or `google-http-client`'s `com.google.api.client.http.apache.*` classes are being used is in `GoogleApacheHttpTransport` class - which is not being consumed in library's code.  

For instantiating transport, `Utils.getDefaultTransport()` is being used - which creates an instance of `NetHttpTransport` (this class uses `java.net.*` network APIs).  

To summarize, it seems that the following exclude suggestion is safe, since this library indeed doesn't use any `org.apache.*` classes:  
```
    implementation('com.google.api-client:google-api-client-android:1.30.9') {
        exclude group: 'org.apache.httpcomponents'
    }
```

As I'm writing this comment, I've noticed that exact `exclude` instructions are written [here](https://googleapis.github.io/google-api-java-client/setup.html) - which makes my whole issue irrelevant.

The issue can be closed, the only thing I want to bring to your attention is that originally, I came across this library's setup instructions via [this guide](https://developers.google.com/android/guides/http-auth) (which doesn't have any `exclude` instructions). Any idea who's responsible for this guide?
